etl/steps/data/garden/faostat/2023-02-22/detected_anomalies.py

```diff
@@ -0,0 +1,358 @@
+"""Detected anomalies.
+
+This module contains a class for each data anomaly detected. The class has the following methods and properties:
+* description: A human-readable text that describes the anomaly.
+  NOTE: The description will be added to the dataset metadata description, and hence will be shown in grapher.
+* checks: A method that ensures the anomaly exists in the data.
+  This is useful to detect if an anomaly has been corrected after a data update.
+* fix: A method that removes the anomaly.
+* inspect: An optional method that plots (in the browser) a visualization that shows the anomaly.
+  It can be used before and after removing the anomalies.
+* handle_anomalies: A helper method that uses all the previous methods in the usual order.
+
+If after an update the anomaly is no longer in the original data, remove the corresponding class.
+
+"""
+import abc
+import os
+from typing import Tuple
+
+import pandas as pd
+import plotly.express as px
+from structlog import get_logger
+
+log = get_logger()
+
+# If environment variable INSPECT_ANOMALIES is set to True, run the step in interactive mode.
+INSPECT_ANOMALIES = bool(os.getenv("INSPECT_ANOMALIES", False))
+
+
+class DataAnomaly(abc.ABC):
+    """Abstract class for data anomaly."""
+
+    def __init__(self) -> None:
+        pass
+
+    @property
+    @abc.abstractmethod
+    def description(self) -> str:
+        # Description of the anomaly, which will be added to the dataset description.
+        pass
+
+    @abc.abstractmethod
+    def check(self, df: pd.DataFrame) -> None:
+        pass
+
+    @abc.abstractmethod
+    def fix(self, df: pd.DataFrame) -> pd.DataFrame:
+        pass
+
+    def inspect(self, df: pd.DataFrame) -> None:
+        pass
+
+    def handle_anomalies(self, df: pd.DataFrame, inspect_anomalies: bool = INSPECT_ANOMALIES) -> pd.DataFrame:
+        log.info(f"Handling anomaly: {self.description}")
+        log.info("Checking that known data anomalies are present in the data")
+        self.check(df=df)
+
+        if inspect_anomalies:
+            log.info("Inspect anomaly before fixing.")
+            self.inspect(df=df)
+
+        log.info("Fixing anomalies.")
+        df_fixed = self.fix(df=df)
+
+        if inspect_anomalies:
+            log.info("Inspect anomaly after fixing.")
+            self.inspect(df=df_fixed)
+
+        return df_fixed
+
+
+def _split_long_title(text: str) -> str:
```

Comment:
Maybe add a short summary of this function? Something like "Formats a given text to have at most 100 characters per line. Line breaks are implemented with the HTML tag `<br>`.